test/samples/structs/structs.rs

```diff
@@ -46,61 +48,81 @@ fn structs(struct_: Struct1, n: i32, struct_list: Vec<Struct1>, triangle: Vec<Po
 }
 
 fn main() {
-    let struct_: Struct1 = read_struct_struct_1();
-    let n: i32 = read_line().parse().unwrap();
-    let mut struct_list: Vec<Struct1> = Vec::with_capacity(n as usize);
-    for _ in 0..n {
-        struct_list.push(read_struct_struct_1());
-    }
-    let mut triangle: Vec<Point> = Vec::with_capacity(3 as usize);
-    for _ in 0..3 {
-        let triangle_elem: Point = read_struct_point();
-        triangle.push(triangle_elem);
-    }
-    let struct_chars: Chars = read_struct_chars();
+    let mut buffer = String::new();
+
+    let struct_ = read_line(&mut buffer)
+        .parse()
+        .expect("invalid `struct` parameter");
+
+    let n = read_line(&mut buffer)
+        .parse()
+        .expect("invalid `n` parameter");
+
+    let struct_list = (0..n)
```

Comment:
The type is resolved during the function call, which explicitly requires a `Vec<Struct1>`, do you think it can be very confusing for a beginner or is it just a question you ask yourself as a reviewer?

Reply:
No it’s probably not confusing. For someone that does not know much it may look like magic, but that’s not a real problem.